### ISSUE TITLE
Fix TestHttps.test_https

### ISSUE THREAD

**Issue by mgedmin** [linkchecker/linkchecker#18]
Currently this test is disabled (with `@pytest.mark.xfail`) because it fails. If I remove the pytest mark and run it, I get:

```
$  tox -- tests/checker/test_https.py 
...
=================================== FAILURES ===================================
_____________________________ TestHttps.test_https _____________________________

args = (<tests.checker.test_https.TestHttps testMethod=test_https>,)
kwargs = {}

    def newfunc (*args, **kwargs):
        if not testfunc():
            pytest.skip("%s is not available" % name)
>       return func(*args, **kwargs)

tests/__init__.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/test_https.py:47: in test_https
    self.direct(url, resultlines, recursionlevel=0, confargs=confargs)
tests/checker/__init__.py:228: in direct
    self.fail_unicode(unicode(os.linesep).join(l))
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: Differences found testing https://www.amazon.com/
E   @@ -1,5 +1,4 @@
E    url https://www.amazon.com/
E    cache key https://www.amazon.com/
E    real url https://www.amazon.com/
E   -info Access denied by robots.txt, checked only syntax.
E    valid
=========================== 1 failed in 1.36 seconds ===========================
ERROR: InvocationError: '/home/mg/src/linkchecker/.tox/py27/bin/py.test tests/checker/test_https.py'
___________________________________ summary ____________________________________
ERROR:   py27: commands failed
```

**Comment by keunes** [comment]
Might this the reason I am seeing the "Result Error: Error: []" in the report for https (ssl) links?

**Comment by mgedmin** [comment]
This failure is not because HTTPS is broken, it's because our test suite tries to test against Amazon.com, which publishes a robots.txt policy forbidding link crawlers like linkchecker from crawling their site.

We need to fix the test suite to use a different HTTPS site for testing (GitHub maybe?).

**Comment by cjmayo** [comment]
Fixed in dee4be4b1d.
